x-pack/plugins/ml/public/application/components/custom_urls/custom_url_editor/utils.ts

```diff
@@ -252,12 +254,19 @@ async function buildDashboardUrlFromSettings(settings: CustomUrlSettings): Promi
   }
 
   const dashboard = getDashboard();
+  let customStart;
```

Comment:
these two new blocks of code are identical, could they be moved to a small function that takes in `settings`?
it could even incorporate the `customStart ?? '$earliest$'` check later and just return `from` and `to`